### DISCUSSION TITLE
How to brake UAV during high speed flight with pixhawk ？

### DISCUSSION THREAD

**Discussion by kalading2** [microsoft/AirSim#4552] [Support Q&A]
When I switch to hover mode, the brake will have inertia, and the UAV can't brake immediately. How to achieve this.

**Comment by ahmed-elsaharti** [comment]
@love-mj Not sure if I understand the question. But if you want the drone to just instantaneously stop on cue while flying at a high speed, I'm not sure if that's entirely doable just because of the way the simulated physics work (since realistically the drone can't not have inertia).
If that's what you're trying to do, you could always try spawning an object in UE4 right in front of the drone on cue to make it instantaneously collide and stop. I don't think the PX4 autopilot (HITL or SITL) would entirely like that though.

**Comment by kalading2** [comment]
@ahmed-elsaharti I've tried the way you said. The attitude of UAV is not what I want . 
 I want to simulate dji UAV avoiding obstacles and hovering when encountering obstacles , I don't know how ？

**Comment by ahmed-elsaharti** [comment]
@love-mj I'm not sure I understand what you're trying to do exactly

Edit: on re-reading the edited comment I think I somewhat get what you're trying to do. To have the quad avoid obstacles I think you would need to write your own piece of code to use the sensor data (maybe lidar) through the APIs to try detect an obstacle and slow down or even hover before collision.
Have a look at the sample python scripts in the repo  (https://github.com/microsoft/AirSim/tree/master/PythonClient/multirotor) and the APi documentation (https://microsoft.github.io/AirSim/apis/) they might help.

**Comment by kalading2** [comment]
@ahmed-elsaharti What you understand is right. But I have tried many methods, I don't know what function to achieve, how to solve it ？
![Blocks - Unreal Editor 2020_11_30 17_02_22](https://user-images.githubusercontent.com/37679671/100591343-923f4980-3330-11eb-8d41-2111ca39b35f.png)

**Comment by lovettchris** [comment]
If you tell it to fly to the new position away from the wall PX4 will do a pretty aggressive banked turn.  If you also tell it to fly back in the exactly direction it was coming from PX4 will do a steep banked stop to change directions.  You might be able to configure PX4 to be more aggressive by increasing these limits in the [PX4 parameters](https://docs.px4.io/master/en/advanced_config/parameter_reference.html.  Basically check all the `MC_*_MAX` and the `MPC_*_MAX` parameters.  I have noticed some of these get reset from time to time, so you can also add these overrides to the AirSim Settings.json file as shown [here](https://microsoft.github.io/AirSim/px4_sitl/).
